src/Symfony/Component/Security/Core/Authentication/AuthenticationProviderManager.php

```diff
@@ -105,6 +106,10 @@ public function authenticate(TokenInterface $token)
                 $this->eventDispatcher->dispatch(new AuthenticationSuccessEvent($result), AuthenticationEvents::AUTHENTICATION_SUCCESS);
             }
 
+            if ($user = $result->getUser() instanceof UserInterface && !method_exists($result->getUser(), 'getUserIdentifier')) {
+                trigger_deprecation('symfony/security-core', '5.3', 'Not implementing method "getUserIdentifier(): string" in user class "%s" is deprecated. This method will replace "getUsername()" in Symfony 6.0.', \get_class($result->getUser()));
```

Comment:
`get_class()` should be replaced by `get_debug_type()` everywhere

Reply:
Do you know the policy here? I know for sure that `$user` is an object (it passed an `instanceof` check), so `get_debug_type()` doesn't add anything.

Reply:
@wouterj It does add proper rendering of anonymous classes. So if you want to fetch an object's class for the sole purpose of printing it, `get_debug_type()` is the safer way.